### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by someaddons**
Closes #
Closes #

# Changes proposed in this pull request
- Fix jobless citizen not stopping when interacted with
- Add entity data debug by hitting with the schematic analyzer in creative
- Entities which have colonyID == 0 now get discarded for bad data
- Reduced lag of the navigator

## Testing
- [x ] Yes I tested this before submitting it.
- [ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java [hunk 1]
```diff
@@ -161,6 +167,12 @@ public void onSyncDataUpdate(final EntityDataAccessor<?> data)
         }
     }
 
+    @Override
+    public boolean registered()
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java:171]: what does this do?

File: src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java [hunk 2]
```diff
@@ -134,6 +134,12 @@ public void updateColonyClient()
                 colonyId = citizen.getEntityData().get(DATA_COLONY_ID);
             }
 
+            if (colonyId == 0)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java:137]: Did you make sure these things don't happen atm regularily "just fine" ?

File: src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java [hunk 3]
```diff
@@ -297,6 +297,7 @@ private ICitizenData spawnCitizenOnPosition(
 
             colony.getEventDescriptionManager().addEventDescription(new CitizenSpawnedEvent(spawnPoint, citizenData.getName()));
         }
+
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java:301]: line?

File: src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java (hunk 2 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java:137]: ye tested a colony, this doesnt get hit normally

File: src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java:171]: just an accessor for the value, used in debug info

File: src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java [hunk 4]
```diff
@@ -297,6 +298,16 @@ private ICitizenData spawnCitizenOnPosition(
...
+        if (world instanceof ServerLevel serverLevel)
+        {
+            final Entity existing = serverLevel.getEntity(citizenData.getUUID());
+            if (existing != null)
+            {
+                existing.discard();
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java:307]: I guess we should log this too

File: src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java:171]: ah ok

File: src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java (hunk 3 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java:301]: you're a line!

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java:301]: No, lines are only 2 dimensional

File: src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java (hunk 4 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/colony/managers/CitizenManager.java:307]: not sure there may be some valid cases where we just re-spawn an entity

File: src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenColonyHandler.java:137]: Should probably add some logging here then when we do this

File: src/main/java/com/minecolonies/core/event/EventHandler.java [hunk 5]
```diff
@@ -143,6 +143,11 @@ public static void onEntityAdded(@NotNull final EntityJoinLevelEvent event)
             {
                 event.setCanceled(true);
             }
+
+            if (event.getEntity() instanceof EntityCitizen citizen && citizen.getCitizenColonyHandler().getColonyId() == 0)
+            {
+                event.setCanceled(true);
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/event/EventHandler.java:150]: and maybe logging here too.

**Pull request merged**